packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -69,222 +107,325 @@ abstract class FlutterGoldenFileComparator extends GoldenFileComparator {
     await goldenFile.writeAsBytes(imageBytes, flush: true);
   }
 
+  @override
+  Uri getTestUri(Uri key, int version) => key;
+
   /// Calculate the appropriate basedir for the current test context.
   @protected
   @visibleForTesting
-  static Directory getBaseDirectory(GoldensClient goldens, LocalFileComparator defaultComparator) {
-    final FileSystem fs = goldens.fs;
+  static Directory getBaseDirectory(LocalFileComparator defaultComparator, Platform platform) {
+    const FileSystem fs = LocalFileSystem();
+    final Directory flutterRoot = fs.directory(platform.environment[_kFlutterRootKey]);
+    final Directory comparisonRoot = flutterRoot.childDirectory(
+      fs.path.join(
+        'bin',
+        'cache',
+        'pkg',
+        'skia_goldens',
+      )
+    );
     final Directory testDirectory = fs.directory(defaultComparator.basedir);
-    final String testDirectoryRelativePath = fs.path.relative(testDirectory.path, from: goldens.flutterRoot.path);
-    return goldens.comparisonRoot.childDirectory(testDirectoryRelativePath);
+    final String testDirectoryRelativePath = fs.path.relative(
+      testDirectory.path,
+      from: flutterRoot.path,
+    );
+    return comparisonRoot.childDirectory(testDirectoryRelativePath);
   }
 
   /// Returns the golden [File] identified by the given [Uri].
   @protected
   File getGoldenFile(Uri uri) {
-    assert(basedir.scheme == 'file');
     final File goldenFile = fs.directory(basedir).childFile(fs.file(uri).path);
-    assert(goldenFile.uri.scheme == 'file');
     return goldenFile;
   }
+
+  /// Prepends the golden Uri with the library name that encloses the current
+  /// test.
+  Uri addPrefix(Uri golden) {
```

Comment:
Does this have to be public? Sounds like this is something that would only get called from within this class and maybe subclasses?